### PR TITLE
[v1.14] Always migrate cilium_calls_* during ELF load

### PR DESCRIPTION
**Pull request by ti-mo**
Manual backport of https://github.com/cilium/cilium/pull/28740.

### CODE REVIEW

**Comment by ti-mo** [comment]
/test-backport-1.14

**Comment by margamanterola** [comment]
Marking as "Don't merge", we'll let the changes bake in main for a few days to check that they are safe.

**Comment by ti-mo** [comment]
/test-backport-1.14

**Comment by margamanterola** [comment]
Can we get this one merged? This would allow to see if it makes things better with the upgrade/downgrade from main test.

**Review by dylandreimerink — APPROVED** [review]

File: pkg/bpf/bpffs_migrate.go [hunk 1]
```diff
@@ -90,35 +92,52 @@ func repinMap(bpffsPath string, name string, spec *ebpf.MapSpec) error {
 	if err != nil {
 		return fmt.Errorf("map not found at path %s: %v", name, err)
 	}
+	defer pinned.Close()
```
Comment (brb) [inline pkg/bpf/bpffs_migrate.go:95]: I assume that closing a pinned map FD doesn't cause any side-effects.

File: pkg/bpf/bpffs_migrate.go [hunk 2]
```diff
@@ -90,35 +92,52 @@ func repinMap(bpffsPath string, name string, spec *ebpf.MapSpec) error {
...
 
 	if pinned.Type() == spec.Type &&
 		pinned.KeySize() == spec.KeySize &&
 		pinned.ValueSize() == spec.ValueSize &&
 		pinned.Flags() == spec.Flags &&
 		pinned.MaxEntries() == spec.MaxEntries {
-		return nil
+		// cilium_calls_xdp is shared between XDP interfaces and should only be
```
Comment (brb) [inline pkg/bpf/bpffs_migrate.go:102]: Could you create a GH issue to document the fact that the `cilium_calls_xdp` is shared among all XDP progs. I think it's a ticking bomb once we move more functionality to the XDP layer.

File: pkg/bpf/bpffs_migrate.go (hunk 1 above)
Comment (ti-mo) [reply pkg/bpf/bpffs_migrate.go:95]: Yep, this Close() should've always been there. Without it, the GC closes the fd.

**Review by brb — COMMENTED** [review]
Thanks, LGTM. Minor comments. I will ACK once https://github.com/cilium/cilium/pull/29045 has passed (ci-ipsec-upgrade: this PR branch -> main -> this PR branch).

File: pkg/bpf/bpffs_migrate.go (hunk 2 above)
Comment (ti-mo) [reply pkg/bpf/bpffs_migrate.go:102]: I was already working on per-device bpffs dirs before picking up this issue, but with https://github.com/cilium/cilium/issues/20691 closed there's indeed nothing left to track. Created https://github.com/cilium/cilium/issues/29047 for this.

**Review by brb — APPROVED** [review]

**Review by squeed — APPROVED** [review]

**Pull request merged**